discovery_engine_core/tooling/bin/clean_sources.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::{
+    fs::{remove_file, File},
+    io::{BufRead, BufReader, BufWriter, Write},
+    path::Path,
+};
+
+use anyhow::{bail, Result};
+use csv::{QuoteStyle, ReaderBuilder, WriterBuilder};
+
+fn main() -> Result<()> {
+    let in_file = "sources_filled.csv";
+    let tmp_file = "sources_fixed.csv";
+    let out_file = "sources_cleaned.csv";
+
+    fix_sources(in_file, tmp_file)?;
+    clean_sources(tmp_file, out_file)?;
+    check_sources(out_file)?;
+    remove_file(tmp_file)?;
+
+    Ok(())
+}
+
+/// Fixes illegal linebreaks in the sources csv file.
+fn fix_sources(in_file: impl AsRef<Path>, out_file: impl AsRef<Path>) -> Result<()> {
+    let reader = BufReader::new(File::open(in_file)?);
+    let mut writer = BufWriter::new(File::create(out_file)?);
+
+    let mut broken_line = String::new();
+    for line in reader.lines() {
+        let line = line?;
+        if line.split(';').count() == 4 {
+            writer.write_all(line.as_bytes())?;
+            writer.write_all(&[b'\n'])?;
+        } else {
+            broken_line.push_str(&line);
```

Comment:
🔧 It looks like that the illegal linebreaks mostly occur in the name of the record.
e.g.

```
Aktuelles Programm
- SWR Fernsehen;swrfernsehen.de;False;47911
```

The problem is that the next line contains three `;` so we only keep `- SWR Fernsehen;swrfernsehen.de` and `broken_line` will just append the lines that contain the linebreak.

Reply:
good catch 👍 i'll change it such that we merge consecutive lines if there have been broken lines before, this should fit the current issue.